lib/Dialect/Datapath/DatapathOps.cpp

```diff
@@ -0,0 +1,97 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements datapath ops.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/Datapath/DatapathOps.h"
+
+using namespace circt;
+using namespace datapath;
+
+LogicalResult CompressOp::verify() {
+  // The compressor must reduce the number of operands by at least 1 otherwise
+  // it fails to perform any reduction.
+  if (getNumOperands() < 3)
+    return emitOpError("Requires 3 or more arguments - otherwise use add");
+
+  if (getNumResults() >= getNumOperands())
+    return emitOpError("Must reduce the number of operands by at least 1");
+
+  if (getNumResults() < 2)
+    return emitOpError("Must produce at least 2 results");
```

Comment:
I think the consensus is that these error strings shouldn't start with an uppercase letter (because it's not the start of a sentence and not even the start of the whole error message)